ecs-cli/modules/utils/compose/ecs_params_reader_test.go

```diff
@@ -512,6 +512,65 @@ task_definition:
 	}
 }
 
+func TestReadECSParams_WithServiceDiscoveryAllFields(t *testing.T) {
+	ecsParamsString := `version: 1
+run_params:
+  service_discovery:
+    container_name: nginx
+    container_port: 80
+    private_dns_namespace:
+      vpc: vpc-8BAADF00D
+      id: ns-CA15CA15CA15CA15
+      name: corp
+      description: This is a private namespace
+    public_dns_namespace:
+      id: ns-C0VF3F3
+      name: amazon.com
+    service_discovery_service:
+      name: mysds
+      description: This is an SDS
+      dns_config:
+        type: A
+        ttl: 60
+      routing_policy: MULTIVALUE
+      healthcheck_custom_config:
+        failure_threshold: 1`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+
+	if assert.NoError(t, err) {
+		serviceDiscovery := ecsParams.RunParams.ServiceDiscovery
+		assert.Equal(t, "nginx", serviceDiscovery.ContainerName, "Expected ContainerName to match")
+		assert.Equal(t, aws.Int64(80), serviceDiscovery.ContainerPort, "Expected ContainerPort to match")
```

Comment:
If we're going to go through the trouble of having an `*int64`, could we perhaps add a test to assert null on this field if not specified? Would also be good to have tests that document expected inputs -- e.g. what makes sense for SRV vs A records.

Reply:
I like this idea re: testing inputs; sounds like an opportunity for a table driven test and runs through a bunch of valid (and invalid) scenarios

Reply:
I don't really agree that there's a need to test if the field is nil if its not specified. After all, this test is already a bit superfluous- its basically testing the yaml parsing library the (yaml parsing library has its own tests already). This isn't like the Healthcheck feature which had custom parsing logic. 

> Would also be good to have tests that document expected inputs -- e.g. what makes sense for SRV vs A records.

I agree there should be documentation of that, but I don't think this is the best place for it. 

There will be code that handles what inputs make sense together (pairings of network modes and DNS Records), but that will come in the next PR and it won't be placed in the ECS Params package. The units tests for that will come in `cli/servicediscovery` package.

Reply:
I agree it makes more sense to test the SD inputs there; I'm guess that's where the real validation logic will be